nifi-nar-bundles/nifi-extension-utils/nifi-event-transport/src/test/java/org/apache/nifi/event/transport/netty/SocketByteArrayMessageDecoderTest.java

```diff
@@ -0,0 +1,166 @@
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.SSLPeerUnverifiedException;
+import javax.net.ssl.SSLSession;
+import javax.security.auth.x500.X500Principal;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.net.InetSocketAddress;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+@ExtendWith(MockitoExtension.class)
+public class SocketByteArrayMessageDecoderTest extends SocketByteArrayMessageDecoder {
+
+    private InetSocketAddress inetSocketAddress;
+    private X509Certificate certificate;
+    @Mock
+    private ChannelHandlerContext channelHandlerContext;
+    @Mock
+    private Channel channel;
+    @Mock
+    private ChannelPipeline channelPipeline;
+    @Mock
+    private Iterator<Map.Entry<String, ChannelHandler>> iterator;
+    @Mock
+    private Map.Entry<String, ChannelHandler> pipelineEntry;
+    @Mock
+    private SslHandler sslHandler;
+    @Mock
+    private ChannelOutboundHandlerAdapter nonSslHandler;
+    @Mock
+    private SSLEngine sslEngine;
+    @Mock
+    private SSLSession sslSession;
+
+    @BeforeEach
+    public void init() {
+        inetSocketAddress = new InetSocketAddress("localhost", 21000);
+        Mockito.when(channelHandlerContext.channel()).thenReturn(channel);
+        Mockito.when(channel.remoteAddress()).thenReturn(inetSocketAddress);
+        Mockito.when(channel.pipeline()).thenReturn(channelPipeline);
+        Mockito.when(channelPipeline.iterator()).thenReturn(iterator);
+        Mockito.when(iterator.hasNext()).thenReturn(true).thenReturn(true).thenReturn(true).thenReturn(false);
+        Mockito.when(iterator.next()).thenReturn(pipelineEntry);
+    }
+
+    @Test
+    public void testDecodeMessageSsl() throws Exception {
+        String expectedSubjectDN = "CN=Subject,O=Subject Organization";
+        String expectedIssuerDN = "CN=Issuer,O=Issuer Organization";
+
+        initSsl();
+
+        List<Object> decoded = new ArrayList<>(1);
+        decode(channelHandlerContext, "test".getBytes(), decoded);
+
+        X500Principal actualSubject = ((ByteArrayMessage)decoded.get(0)).getSslSessionStatus().getSubject();
+        X500Principal actualIssuer = ((ByteArrayMessage)decoded.get(0)).getSslSessionStatus().getIssuer();
+        assertEquals(expectedSubjectDN, actualSubject.getName());
+        assertEquals(expectedIssuerDN, actualIssuer.getName());
+    }
+
+    @Test
+    public void testDecodeMessageNoSsl() {
+        initNoSslHandler();
+
+        List<Object> decoded = new ArrayList<>(1);
+        decode(channelHandlerContext, "test".getBytes(), decoded);
```

Comment:
The `"test".getBytes()` method should be declared statically, and `getBytes()` should include the Standard UTF-8 character set.